sympy/utilities/misc.py

```diff
@@ -355,7 +355,8 @@ def translate(s, a, b=None, c=None):
 
     mr = {}
     if a is None:
-        assert c is None
+        if c is not None:
+            raise TypeError('Called translate with 4 arguments and a second argument of None, expected 3 arguments if second argument is None')
```

Comment:
For my own edification, I just grepped SymPy for lines longer than 80 characters and came up with quite a few.  If you're curious, just run `grep -r --include="*.py" ".\{80\}" .` from the SymPy project root.